### PR TITLE
examples/gnrc_networking: add explanation for RIOT<->RIOT communication

### PR DESCRIPTION
**Pull request by Lotterleben**
Add some instructions on how to make RIOTs talk amongst each other to the `gnrc_networking` example.

### CODE REVIEW

File: examples/gnrc_networking/README.md [hunk 1]
```diff
@@ -63,3 +67,53 @@ Now, on the RIOT side, send a UDP packet using:
...
 
 You should see `testmessage` appear in netcat. Instead of using netcat, you can of course write your own software, but you may have to bind the socket to a specific interface (tap0 in this case). For an example that shows how to do so, see [here](https://gist.github.com/backenklee/dad5e80b764b3b3d0d3e).
+
+## Connecting two RIOT instances
+
+When using native (i.e. when you're trying this on your Linux machine), you first need to set up two tap devices and a bridge that connects them. This constitutes a virtual network that the RIOT instances can use to communicate.
+
+    ./../../dist/tools/tapsetup/tapsetup --create 2
+
+Then, make sure you've compiled the application by calling ``make` and start the first RIOT instance by invoking `make term`. In the RIOT shell, get to know the IP address of this node:
```
Comment (cgundogan) [inline examples/gnrc_networking/README.md]: one '`' too much

File: examples/gnrc_networking/README.md [hunk 2]
```diff
@@ -63,3 +67,53 @@ Now, on the RIOT side, send a UDP packet using:
...
+
+This node is now ready to receive data on port `8808`.
+
+In a second terminal, start a second RIOT instance, this time listening on `tap1`:
+
+    sudo bin/native/gnrc_networking.elf tap1
+
+In the RIOT shell, you can now send a message to the first RIOT:
+
+    udp send fe80::ccf5:e1ff:fec5:f75 8808 testmessage
```
Comment (cgundogan) [inline examples/gnrc_networking/README.md]: `>` is missing

File: examples/gnrc_networking/README.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # gnrc_networking example
 
+You can try out the code in two different ways: You can either communicate between the RIOT machine and its Linux host, or you can communicate between two RIOT instances.
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: You should point out, that the RIOT<->Linux case is just for native.

Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: Good point. Will do.

Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: Good point. Will do.

File: examples/gnrc_networking/README.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # gnrc_networking example
 
+You can try out the code in two different ways: You can either communicate between the RIOT machine and its Linux host, or you can communicate between two RIOT instances. Note that the former only works with native, i.e. if you run RIOT on your Linux machine.
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: Line-length

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: Oh and the wording is a little weird… Communication with a non-native node and Linux is possible through Ethernet, just not in the way it is descriped here.

File: examples/gnrc_networking/README.md [hunk 5]
```diff
@@ -18,13 +20,15 @@ First, create a tap interface:
 Now you can start the `gnrc_networking` example by invoking `make term`. This should automatically connect to the `tap0` interface. If
 this doesn't work for some reason, run `make` without any arguments, and then run the binary manually like so (assuming you are in the `examples/gnrc_networking` directory):
 
+    sudo bin/native/gnrc_networking.elf tap0
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: Isn't this what `make term` is for? (with that you can also drop the remark about Linux<->RIOT communication above)

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: also: should not need sudo

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: > (with that you can also drop the remark about Linux<->RIOT communication above)

Ah nope, the text still refers to `tap0`

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: > also: should not need sudo

if you run `tapsetup` without sudo https://github.com/RIOT-OS/RIOT/blob/master/dist/tools/tapsetup/tapsetup#L104 should ensure the running user has access to the tap0

Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: I'm a bit puzzled now... just delete the `sudo`, correct?

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: just write `make term` ;-)

Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: But... The entire purpose of the text above is to instruct readers what to do if make term *doesn't* work as expected?

File: examples/gnrc_networking/README.md [hunk 6]
```diff
@@ -62,4 +81,63 @@ Now, on the RIOT side, send a UDP packet using:
...
+interface (tap0 in this case). For an example that shows how to do so, see
+[here](https://gist.github.com/backenklee/dad5e80b764b3b3d0d3e).
+
+## Connecting two RIOT instances
+
+When using native (i.e. when you're trying this on your Linux machine), you first
+need to set up two tap devices and a bridge that connects them. This constitutes a
+virtual network that the RIOT instances can use to communicate.
+
+    ./../../dist/tools/tapsetup/tapsetup --create 2
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: `--create 2` is the default setting, but maybe it's useful for people to see that they *can* create a number of interfaces that is different from 2

Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: Yup, that was the idea :) I thought that adding a note would make the whole thing even lengthier so I opted to do it this way...

File: examples/gnrc_networking/README.md (hunk 5 above)
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: Why shouldn't it? Either run `make term` or `PORT=tap1 make term`. If this doesn't work it's a bug in RIOT and should be fixed...

File: examples/gnrc_networking/README.md (hunk 6 above)
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: :+1:

File: examples/gnrc_networking/README.md (hunk 5 above)
Comment (Lotterleben) [inline examples/gnrc_networking/README.md]: Ha, see, I didn't know that `PORT=tap1 make term` was a thing :D I'll just add that then? I'm not particularly invested in this entire paragraph, I just saw that there was a line missing and I tried to fill it out ;)

**Comment by Lotterleben** [comment]
What do you think? Better this way? :)

File: examples/gnrc_networking/README.md [hunk 7]
```diff
@@ -1,38 +1,54 @@
...
 
     sudo ip tuntap add tap0 mode tap user ${USER}
     sudo ip link set tap0 up
 
-Now you can start the `gnrc_networking` example by invoking `make term`. This should automatically connect to the `tap0` interface. If
-this doesn't work for some reason, run `make` without any arguments, and then run the binary manually like so (assuming you are in the `examples/gnrc_networking` directory):
+Now you can start the `gnrc_networking` example by invoking `make term`. This should
+automatically connect to the `tap0` interface. If this doesn't work for some reason,
+run `make` without any arguments, and then run the binary manually like so (assuming
+you are in the `examples/gnrc_networking` directory):
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: This last sentence does not make that much sense.

Comment (authmillenon) [inline examples/gnrc_networking/README.md]: If this doesn't work for any reason, run `make term` with the `tap0` interface as the `PORT` environment variable:

**Comment by Lotterleben** [comment]
addressed comments.. should I squash? :)

File: examples/gnrc_networking/README.md [hunk 8]
```diff
@@ -62,4 +80,63 @@ Now, on the RIOT side, send a UDP packet using:
...
+
+    > udp server start 8808
+
+This node is now ready to receive data on port `8808`.
+
+In a second terminal, start a second RIOT instance, this time listening on `tap1`:
+
+    PORT=tap1 make term
+
+In the RIOT shell, you can now send a message to the first RIOT:
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: *first RIOT instance

File: examples/gnrc_networking/README.md [hunk 9]
```diff
@@ -62,4 +80,63 @@ Now, on the RIOT side, send a UDP packet using:
...
+
+    > udp send fe80::ccf5:e1ff:fec5:f75 8808 testmessage
+
+*(Make sure to copy the actual
+[link-local address](https://en.wikipedia.org/wiki/Link-local_address) of your first
+RIOT into the above command)*
```
Comment (authmillenon) [inline examples/gnrc_networking/README.md]: First RIOT instance

**Comment by authmillenon** [comment]
sorry, but with doc PRs I'm very nitpicky ^^

**Comment by Lotterleben** [comment]
no sorries, that's great! better now? :)

**Comment by authmillenon** [comment]
Please squash

**Comment by Lotterleben** [comment]
Squishy...Squashy...Squashed. :)

**Comment by cgundogan** [comment]
Looks also good here. ACK and GO once travis agrees

**Comment by cgundogan** [comment]
no need to wait for travis here. Changes are only in the README.md => GO

**Pull request merged**